scipy/linalg/_basic.py

```diff
@@ -51,23 +51,26 @@ def _solve_check(n, info, lamch=None, rcond=None):
 
 def _find_matrix_structure(a):
     n = a.shape[0]
-    below, above = bandwidth(a)
-
-    if below == above == 0:
-        return 'diagonal'
-    elif above == 0:
-        return 'lower triangular'
-    elif below == 0:
-        return 'upper triangular'
-    elif above <= 1 and below <= 1 and n > 3:
-        return 'tridiagonal'
-
-    if np.issubdtype(a.dtype, np.complexfloating) and ishermitian(a):
-        return 'hermitian'
+    n_below, n_above = bandwidth(a)
+
+    if n_below == n_above == 0:
+        kind = 'diagonal'
+    elif n_above == 0:
+        kind = 'lower triangular'
+    elif n_below == 0:
+        kind = 'upper triangular'
+    elif n_above <= 1 and n_below <= 1 and n > 3:
+        kind = 'tridiagonal'
+    elif n_above <= n/10 and n_below <= n/10:
+        kind = 'banded'
```

Comment:
```suggestion
    # very conservative condition to avoid the possibility of slowdown
    elif (n > 100) and (n_above <= 5) and (n_below <= 5):
        kind = 'banded'
```

Reply:
This is not going to happen to the same number of bands. The issue is if the user does not know the band structure, then some arrays will take x amount of time and some will take much longer and it will be very hard to know why it is happening. So this introduces inconsistency in the performance. This is one of the reasons why other software don't do this. Like I mentioned below, you can't find a good threshold to cut it out because there is always another array that misses out the hard conditions. 

In general, these are not automated if there is no algorithmic way to define good bounds. And bandedness is one of those things that folks over time gave up trying because you always have too many edge cases. Folks will think "well I think 10 is a better threshold why is it set to 5" and it's your preference against their preferences with no right answer.

Reply:
OK, so I understand now that the concern is that:
- These two lines cause an undesirable increase in speed in special cases. 
- The increase in speed is undesirable because many users will be confused as to why the performance is significantly better in some cases but not others. 
- This concern does not apply to the improved performance in triangular, tridiagonal, or diagonal matrices because those structures are immediately obvious to these users, but our threshold for whether a matrix is "banded" will not be. (Unless we document it. Which we can do.)

@j-bowhay Thanks for commenting before. I think we're coming to a conclusion here, so LMK which way to go - remove this condition or not.

Reply:
No problem at all. This is already a great addition. I am very happy that finally we have some automation in the structure detection for a lot of cases.

Reply:
I think this is still very useful even if the above condition is removed

Reply:
Ok. I'll remove it in this PR, then add it back in a follow-up along with a line of documentation. There is an obvious place for it.